extensions/ipynb/src/serializers.ts

```diff
@@ -123,7 +123,10 @@ function createCodeCellFromNotebookCell(cell: NotebookCellData, preferredLanguag
 
 	const codeCell: any = {
 		cell_type: 'code',
-		execution_count: cell.executionSummary?.executionOrder ?? null,
+		// Possible the metadata was edited as part of diff view
+		// In diff view we display execution_count as part of metadata, hence when execution count changes in metadata,
+		// We need to change that here as well, i.e. give preference to any execution_count value in metadata.
+		execution_count: cellMetadata.execution_count ?? cell.executionSummary?.executionOrder ?? null,
```

Comment:
Bug fix, reverting metadata or changing exec count in metadata doesn't work even in todays diff viewer